src/main/java/com/efficio/etl/web/ImportObservationsController.java

```diff
@@ -189,6 +189,7 @@ public String confirmImport(
 			importData.setConditions(referenceWorkbook.getConditions());
 			this.dataImportService.addExptDesignVariableIfNotExists(importData, importData.getFactors(), programUUID);
 			this.dataImportService.addLocationIDVariableIfNotExists(importData, importData.getFactors(), programUUID);
+			this.dataImportService.addEntryTypeVariableIfNotExists(importData, importData.getFactors(), programUUID);
```

Comment:
Remove importData.getFactors parameter. Retrieve the factors measurement variables list from importData parameter in the actual method. Also apply this change in the method calls above: addLocationIDVariableIfNotExists and addExptDesignVariableIfNotExists